components/eam/bld/build-namelist

```diff
@@ -915,6 +915,12 @@ if ($rad_pkg eq 'rrtmg' or $chem =~ /waccm_mozart/) {
 # COSP simulator
 if ($cfg->get('cosp')) {
     add_default($nl, 'docosp', 'val'=>'.true.');
+
+    # following serves to always set a default when no matching value found in  namelist_defaults_eam.xml
```

Comment:
Is this extra code necessary? I actually think this can all be replaced by a single line in `namelist_defaults_eam.xml` that sets an overall default value (with no `hgrid` match) for `cosp_ncolumns`, and then overrides for `hgrid=ne1024np4`. I.e.,
```
<cosp_ncolumns                                 > 50 </cosp_ncolumns>
<cosp_ncolumns phys="default" hgrid="ne1024np4"> 10 </cosp_ncolumns>
```

Reply:
Very true, @brhillman , adding an extra entry in namelist_defaults_eam.xml would have equivalent effect.  I tested both before implementing this version. We can simplify it if to use a grid-independent value of 10, as  Peter prefers.

No matter what, the extra call of add_default for cosp_ncolumns is always needed.

Reply:
Ben, if to remove explicit grid dependence, I would inline to keep the changes in build-namelist. The phys='default' will still be used, to differ from use of other potential physics package (such as MMF)

Reply:
Sorry, I should have been more clear. I think you're right that `add_default` is needed, but I don't think this part is:
```
    my $val = undef;
    $val = get_default_value('cosp_ncolumns');
    $val = 50 if (! defined($val) );
```
Most of the rest of the settings in `build-namelist` seem to just call `add_default`.

Reply:
These are still needed, unless we either remove the 'phys' modifider, or add another entry in namelist_defaults_eam.xml to cover all scenarios.